x-pack/plugins/security_solution/public/common/components/filter_bar/index.tsx

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { memo, useMemo, useCallback } from 'react';
+
+import type { DataViewBase, Filter, Query } from '@kbn/es-query';
+import { FilterManager, TimeHistory } from '@kbn/data-plugin/public';
+import { DataView } from '@kbn/data-views-plugin/public';
+import { SearchBar, SearchBarProps } from '@kbn/unified-search-plugin/public';
+import { Storage } from '@kbn/kibana-utils-plugin/public';
+
+export interface FilterBarComponentProps {
+  dataTestSubj?: string;
+  indexPattern: DataViewBase;
+  isLoading?: boolean;
+  isRefreshPaused?: boolean;
+  filterQuery: Query;
+  filterManager: FilterManager;
+  filters: Filter[];
+  refreshInterval?: number;
+  displayStyle?: SearchBarProps['displayStyle'];
+}
+
+export const FilterBar = memo<FilterBarComponentProps>(
+  ({
+    indexPattern,
+    isLoading = false,
+    isRefreshPaused,
+    filterQuery,
+    filterManager,
+    filters,
+    refreshInterval,
+    dataTestSubj,
+    displayStyle,
+  }) => {
+    const onFiltersUpdated = useCallback(
+      (newFilters: Filter[]) => {
+        filterManager.setFilters(newFilters);
+      },
+      [filterManager]
+    );
+
+    const indexPatterns = useMemo(() => [indexPattern], [indexPattern]);
+    const timeHistory = useMemo(() => new TimeHistory(new Storage(localStorage)), []);
```

Comment:
I realised that we don't really need time history in this component at all as we are not using time range bar in this case (`showDatePicker={false}`). I checked the `SearchBar` component and the `timeHistory` prop is required there, though it's subcomponent `QueryBarTopRow` declares this prop as an optional. Do you see any reason for `timeHistory` to be required instead of optional in `SearchBar`? If not then I could just remove it from the new `FilterBar` component.